modules/cancerBiomarkers.py

```diff
@@ -0,0 +1,352 @@
+            cancerBiomarkersEvidenceGenerator.zip_alterations_with_type,
+            ArrayType(ArrayType(StringType()))
+        )
+
+    def main(
+        self,
+        biomarkers_table: str,
+        source_table: str,
+        disease_table: str,
+        drug_index: str,
+        output_file: str
+    ) -> None:
+        """Loads and processes inputs to generate the Cancer Biomarkers evidence strings"""
+
+        # Import data
+        biomarkers_df = self.spark.read.csv(biomarkers_table, sep='\t', header=True)
+        source_df = self.spark.read.json(source_table).select(
+            col('label').alias('niceName'),
+            'source', 'url')
+        disease_df = self.spark.read.json(disease_table).select(
+            regexp_replace(col('name'), '_', '').alias('tumor_type'),
+            regexp_extract(col('url'), '[^/]+$', 0).alias('diseaseFromSourceMappedId'))
+        drugs_df = self.spark.read.parquet(drug_index).select(
+            col('id').alias('drugId'), col('name').alias('drug'))
+
+        # Process inputs to generate evidence strings
+        evidence = self.compute_biomarkers(
+            biomarkers_df, source_df, disease_df, drugs_df
+        )
+
+        # Write evidence strings
+        self.write_evidence_strings(output_file)
+        logging.info(f'{evidence.count()} evidence strings have been saved to {output_file}.')
+
+    def compute_biomarkers(
+        self,
+        biomarkers_df: DataFrame,
+        source_df: DataFrame,
+        disease_df: DataFrame,
+        drugs_df: DataFrame
+    ) -> DataFrame:
+        """The diverse steps to prepare and enrich the input table are computed"""
+
+        biomarkers_enriched = (
+            biomarkers_df
+            .select(
+                'Biomarker', 'IndividualMutation',
+                array_distinct(split(col('Alteration'), ';')).alias('alterations'),
+                array_distinct(split(col('Gene'), ';')).alias('gene'),
+                split(col('AlterationType'), ';').alias('alteration_types'),
+                array_distinct(split(col("PrimaryTumorTypeFullName"), ";")).alias('tumor_type_full_name'),
+                array_distinct(split(col('Drug'), ';')).alias('drug'),
+                'DrugFullName', 'Association', 'EvidenceLevel', 'gDNA',
+                array_distinct(split(col('Source'), ';')).alias('source')
+            )
+            .withColumn('tumor_type_full_name', explode(col('tumor_type_full_name')))
+            .withColumn('tumor_type', translate(col('tumor_type_full_name'), ' -', ''))
+            .withColumn('gene', explode(col('gene')))
+            .withColumn('drug', explode(col('drug')))
+            # Override specific genes
+            .withColumn(
+                'gene',
+                when(col('gene') == 'C15orf55', 'NUTM1')
+                .when(col('gene') == 'MLL', 'KMT2A')
+                .when(col('gene') == 'MLL2', 'KMT2D')
+                .otherwise(col('gene'))
+            )
```

Comment:
I think this is not necessarily the best idea to implement the mapping like this... I can imagine a growing set of genes with old or not canonical names and this pattern is not suitable. But this is very minor comment. And it's also quite deep down... a top level dictionary would be better like `ALTERATIONTYPE2FUNCTIONCSQ`. as you did in line 187 and 188.

Reply:
I think it is not the same case, as, with the dictionary, the idea is that there is a mapping for every single value of the column, otherwise the column will be null. Whereas here we only want to filter and change this handful of gene names.